### ISSUE TITLE
Tests with multi-line names are skipped when using filter

### ISSUE THREAD

**Issue by mantasak** [etorreborre/specs2#764]
I am facing an issue that test cases with multi-line names (e.g. using scala triple quote syntax; but not limited to) are being skipped. This only happens when using a test filter.

Given a test case:

```
import org.specs2.mutable.SpecificationWithJUnit

class QuickStartSpec extends SpecificationWithJUnit {

  "first\tsecond" >> {
    println("I am in tabbed test")
    1 must_== 1
  }

  "first\nsecond" >> {
    println("I am in newlined test")
    1 must_== 1
  }

}
```

The test with tab in its' name runs just fine with:

```
  sbt "testOnly QuickStartSpec -- -ex \Afirst\tsecond\Z"
```

However, I am unable to get the test with new line symbol to run in any way (it is always skipped with pass for the empty test suite).

I tried:

``` 
  sbt "testOnly QuickStartSpec -- -ex \Afirst\nsecond\Z"
```
```
  sbt "testOnly QuickStartSpec -- -ex \Afirst\A\Zsecond\Z"
```
``` 
 sbt "testOnly QuickStartSpec -- -ex \Qfirst\nsecond\E"
```
```
  sbt "testOnly QuickStartSpec -- -ex \Qfirst
  second\E"
```

As mentioned before, exact same behavior happens for tests that use multiline names with triple quote strings.

Can you please confirm if it is an issue with the regex filter, or is it my error?

Would be great to have such a case documented.

**Comment by symbiont-eric-torreborre** [comment]
It is likely to be an issue with the regex filter yes. I'll have a look at that.

**Comment by mantasak** [comment]
@symbiont-eric-torreborre I can help with verifying and/or fixing the issue if you want. 

In such case, I would appreciate if you could point me to the right direction where I can find the regex matching logic.

**Comment by symbiont-eric-torreborre** [comment]
I would love to! 

The selection code starts [here](https://github.com/etorreborre/specs2/blob/master/core/shared/src/main/scala/org/specs2/specification/process/DefaultSelector.scala#L41).

`matchesSafely` is defined [here](https://github.com/etorreborre/specs2/blob/8305db76c5084e4b3ce5827ce23117f6fb6beee4/common/shared/src/main/scala/org/specs2/text/Regexes.scala#L20) and there are some (basic) tests [there](https://github.com/etorreborre/specs2/blob/8305db76c5084e4b3ce5827ce23117f6fb6beee4/core/shared/src/test/scala/org/specs2/text/RegexesSpec.scala)

**Comment by mantasak** [comment]
Thanks for the guidance @symbiont-eric-torreborre.

It was pretty straightforward to find the cause - [link to code](https://github.com/etorreborre/specs2/blob/8305db76c5084e4b3ce5827ce23117f6fb6beee4/common/shared/src/main/scala/org/specs2/text/Regexes.scala#L22).
Opened a pull request, we can continue the discussion there if anything is not as you would expect.

**Comment by genrym** [comment]
Hi @symbiont-eric-torreborre .
Can you say when is the next release planned? We would like to have this fix :)

**Comment by etorreborre** [comment]
Released in `4.7.1`
